service/controller/app/v1/resource/chart/update_test.go

```diff
@@ -0,0 +1,148 @@
+					Name:       "my-cool-prometheus",
+					Namespace:  "monitoring",
+					TarballURL: "https://giantswarm.github.com/app-catalog/kubernetes-prometheus-1.0.1.tgz",
+				},
+			},
+			desiredResource: &v1alpha1.Chart{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Chart",
+					APIVersion: "application.giantswarm.io",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "prometheus",
+					Labels: map[string]string{
+						"app": "prometheus",
+					},
+				},
+				Spec: v1alpha1.ChartSpec{
+					Name:       "my-cool-prometheus",
+					Namespace:  "monitoring",
+					TarballURL: "https://giantswarm.github.com/app-catalog/kubernetes-prometheus-1.0.0.tgz",
+				},
+			},
+			expectedChart: &v1alpha1.Chart{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Chart",
+					APIVersion: "application.giantswarm.io",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "prometheus",
+					Labels: map[string]string{
+						"app": "prometheus",
+					},
+				},
+				Spec: v1alpha1.ChartSpec{
+					Name:       "my-cool-prometheus",
+					Namespace:  "monitoring",
+					TarballURL: "https://giantswarm.github.com/app-catalog/kubernetes-prometheus-1.0.0.tgz",
+				},
+			},
+		},
+		{
+			name: "case 1: chart should not be update",
+			currentResource: &v1alpha1.Chart{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Chart",
+					APIVersion: "application.giantswarm.io",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "prometheus",
+					Labels: map[string]string{
+						"app": "prometheus",
+					},
+				},
+				Spec: v1alpha1.ChartSpec{
+					Name:       "my-cool-prometheus",
+					Namespace:  "monitoring",
+					TarballURL: "https://giantswarm.github.com/app-catalog/kubernetes-prometheus-1.0.0.tgz",
+				},
+			},
+			desiredResource: &v1alpha1.Chart{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Chart",
+					APIVersion: "application.giantswarm.io",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "prometheus",
+					Labels: map[string]string{
+						"app": "prometheus",
+					},
+				},
+				Spec: v1alpha1.ChartSpec{
+					Name:       "my-cool-prometheus",
+					Namespace:  "monitoring",
+					TarballURL: "https://giantswarm.github.com/app-catalog/kubernetes-prometheus-1.0.0.tgz",
+				},
+			},
+			expectedChart: &v1alpha1.Chart{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			var err error
+
+			kc, err := kubeconfigtest.New(nil)
+			if err != nil {
+				t.Fatalf("error == %#v, want nil", err)
+			}
+			c := Config{
+				G8sClient:      fake.NewSimpleClientset(),
+				K8sClient:      k8sfake.NewSimpleClientset(),
+				KubeConfig:     kc,
+				Logger:         microloggertest.New(),
+				WatchNamespace: "default",
+			}
+			r, err := New(c)
+			if err != nil {
+				t.Fatalf("error == %#v, want nil", err)
+			}
+
+			got, err := r.newUpdateChange(context.Background(), tt.currentResource, tt.desiredResource)
+			if err != nil {
+				t.Errorf("Resource.newCreateChange() error = %v", err)
```

Comment:
Why is this `Errorf`? Also the text is wrong.

I think you can just use `t.Fatalf("error == %#v, want nil", err)` here.